receiver/sapnetweaverreceiver/metrics.go

```diff
@@ -0,0 +1,275 @@
+	metricName := "CurrentSecuritySessions"
+	val, err := parseResponse(metricName, alertTreeResponse)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	err = s.mb.RecordSapnetweaverSessionsSecurityCountDataPoint(now, val)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverWorkProcessesActiveCount(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "Total Number of Work Processes"
+	val, err := parseResponse(metricName, alertTreeResponse)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	err = s.mb.RecordSapnetweaverWorkProcessesActiveCountDataPoint(now, val)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverSessionsWebCountDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "Web Sessions"
+	val, err := parseResponse(metricName, alertTreeResponse)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	err = s.mb.RecordSapnetweaverSessionsWebCountDataPoint(now, val)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverSessionsBrowserCountDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "Browser Sessions"
+	val, err := parseResponse(metricName, alertTreeResponse)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	err = s.mb.RecordSapnetweaverSessionsBrowserCountDataPoint(now, val)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverSessionsEjbCountDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "EJB Sessions"
+	val, err := parseResponse(metricName, alertTreeResponse)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	err = s.mb.RecordSapnetweaverSessionsEjbCountDataPoint(now, val)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverIcmAvailabilityDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "ICM"
+	val, ok := alertTreeResponse[metricName]
+	if !ok {
+		errs.AddPartial(1, fmt.Errorf(collectMetricError, metricName, errValueNotFound))
+		return
+	}
+
+	stateColorCode, err := stateColorToInt(models.StateColor(val))
+	if err != nil {
+		errs.AddPartial(1, fmt.Errorf(collectMetricError, metricName, err))
+		return
+	}
+
+	stateColorAttribute, err := stateColorToAttribute(models.StateColor(val))
+	if err != nil {
+		errs.AddPartial(1, fmt.Errorf(collectMetricError, metricName, err))
+		return
+	}
+
+	s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, stateColorCode, stateColorAttribute)
+}
```

Comment:
I think this encoding might be a little off. It looks like both the value of the datapoint and label is set dynamically based on the state color, but I think it should just be one of those two.

We should either:
1. match what the elasticsearch receiver does, and do a 1-hot encoding (that is, always emit a datapoint for each possible label value, and emit a "1" for the active color code, and 0 for all the inactive color codes)
2. Emit a single datapoint with the stateColorCode, and no label

I think 1 is probably ideal, since that's what we've done in the past in OTEL.

Reply:
Added in the 1-hot encoding suggested in point 1.